### PR TITLE
Fixes part of #1267, updated remaining fragment implementations

### PR DESCRIPTION
**Pull request by sajalasati**
## Explanation

In continuation to my earlier PR [#1280](https://github.com/oppia/oppia-android/pull/1280) this fixes the **remaining** part of the the issue [#1267](https://github.com/oppia/oppia-android/issues/1267) (except the `ExplorationFragment` and `ExplorationManagerFragment` mentioned in the issue checklist). Implemented the `newInstance()` function as desired in the issue raised (just a refactoring) and also  updated parts of the code base affected by this new implementation i.e which were using the old implementation - there was only one such file `ViewPagerAdapter`.

Requesting @vinitamurthi @nikitamarysolomanpvt @rt4914 for a review.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@sajalasati Holding this PR for now as #1280 code will be needed in this. Once #1280 is merged in to develop, merge latest develop in this branch.

**Comment by vinitamurthi** [comment]
FYI #1280 is now merged, @sajalasati can you pull the latest changes from develop and update this PR?

File: app/src/main/java/org/oppia/app/topic/ViewPagerAdapter.kt [hunk 1]
```diff
@@ -27,22 +27,13 @@ class ViewPagerAdapter(
         return TopicInfoFragment.newInstance(internalProfileId, topicId)
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/ViewPagerAdapter.kt]: In this file there will some changes:
1. `args` Bundle is not getting used anymore, you can remove that.
2. Also there will be an underline below `when` which should suggesting, `Lift return out of when`, use this option and see the difference.

File: app/src/main/java/org/oppia/app/topic/lessons/TopicLessonsFragment.kt [hunk 2]
```diff
@@ -16,6 +16,23 @@ private const val KEY_CURRENT_EXPANDED_LIST_INDEX = "CURRENT_EXPANDED_LIST_INDEX
 
 /** Fragment that contains subtopic list for lessons mode. */
 class TopicLessonsFragment : InjectableFragment(), ExpandedChapterListIndexListener, StorySummarySelector {
+
+  companion object {
+    /** Returns a new [TopicLessonsFragment] */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/lessons/TopicLessonsFragment.kt]: Add full-stop at the end of the sentence `/** Returns a new [TopicLessonsFragment]. */`

File: app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragment.kt [hunk 3]
```diff
@@ -14,6 +14,18 @@ private const val KEY_SKILL_ID_LIST = "SKILL_ID_LIST"
 
 /** Fragment that displays skills for topic practice mode. */
 class TopicPracticeFragment : InjectableFragment() {
+  companion object {
+    /** Returns a new [TopicPracticeFragment] */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragment.kt]: Add full-stop at the end of the sentence `/** Returns a new [TopicPracticeFragment]. */`

File: app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt [hunk 4]
```diff
@@ -13,6 +13,18 @@ import javax.inject.Inject
 
 /** Fragment that card for topic revision. */
 class TopicRevisionFragment : InjectableFragment(), RevisionSubtopicSelector {
+  companion object {
+    /** Returns a new [TopicRevisionFragment] */
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt]: Add full-stop at the end of the sentence `/** Returns a new [TopicRevisionFragment]. */`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Looks good. Suggested some nit changes.

**Review by vinitamurthi — APPROVED** [review]
LGTM apart from the changes Rajat has mentioned

File: app/src/main/java/org/oppia/app/topic/ViewPagerAdapter.kt (hunk 1 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/ViewPagerAdapter.kt]: Done

File: app/src/main/java/org/oppia/app/topic/lessons/TopicLessonsFragment.kt (hunk 2 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/lessons/TopicLessonsFragment.kt]: Done

File: app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragment.kt (hunk 3 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/practice/TopicPracticeFragment.kt]: Done

File: app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt (hunk 4 above)
Comment (sajalasati) [reply app/src/main/java/org/oppia/app/topic/revision/TopicRevisionFragment.kt]: Done

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**